staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -282,12 +286,11 @@ func ParseWatchCacheSizes(cacheSizes []string) (map[schema.GroupResource]int, er
 		if err != nil {
 			return nil, fmt.Errorf("invalid size of watch cache size: %s", c)
 		}
-		if size < 0 {
```

Comment:
disallow < -1?

Reply:
Yes - you're explicitly using -1.

Reply:
still outstanding. also, it looks like `genericregistry.StorageWithCacher` already skipped setting up caches if the size was 0, can we just use that instead of negative numbers? are we relying on 0 defaulting to `DefaultWatchCacheSize` anywhere?